### DISCUSSION TITLE
What are the zarr heuristics used for chunking in to_zarr()?

### DISCUSSION THREAD

**Discussion by hansmohrmann** [zarr-developers/zarr-python#1069] [Q&A] [answered]
From the [documentation](https://docs.xarray.dev/en/stable/generated/xarray.Dataset.to_zarr.html) in xarray.Dataset.to_zarr(), if no chunks are provided, 

> If neither Dask chunks nor encoding chunks are present, chunks will be determined automatically by Zarr

> If not other chunks are found, Zarr uses its own heuristics to choose automatic chunk sizes

Is there a way to get these chunks _before_ calling to_zarr() (without explicitly assigning chunks)? I'm assuming that this is done by zarr and not xarray.

Thank you very much!

**Comment by d-v-b** [comment, marked as answer]
i believe that logic is inside this function: https://github.com/zarr-developers/zarr-python/blob/e1b71a38ca2643e21dcfab4c277d755cf01f4078/zarr/util.py#L81

**Reply by hansmohrmann** [reply]
Thank you, that's perfect! zarr.util.guess_chunks(da.shape, typesize=da.dtype.itemsize) got me there.
